### ISSUE TITLE
Unhandle promise rejection - chainable populate - bug?

### ISSUE THREAD

**Issue by HenrikGr** [Automattic/mongoose#6129]
I'm using mongoose 5.0.4, node.js 8.9.4 and have a problem that I may think is a bug?
I'm getting an Unhandled promise rejection when Mongoose tries to insert the client via populate in the getAccessToken async function.

I would appreciate any help. 

```
const getAccessToken = async function(accessToken) {
debug('getAccessToken - with accessToken: ' + accessToken );

try {
  const dbToken = await AccessToken.findOne({accessToken })
    .populate('user')
    .populate('client')
    .lean()
    .exec();
  if (!dbToken) {
    debug('getAccessToken: not found');
    return false;
  }
  return dbToken;
  } catch(err) {
    debug('getAccessToken - err', err);
  }
};
```

The log/output/error message I get is this:

```
Mongoose: accesstokens.findOne({ accessToken: '7d4245c444c843dc44463d318dfecdedeb844ced' }, { fields: {} })
Mongoose: users.find({ _id: { '$in': [ ObjectId("581dabef77e0ee0038fd50de") ] } }, { fields: {} })
Mongoose: clients.find({ _id: { '$in': [ ObjectId("5a5a6c5fbc4226c36dfcf5aa") ] } }, { fields: {} })

Mongoose: users.insert({ admin: false, _id: ObjectId("5a82825b82b5ad5ec43b3566"), username: 'henrik@test.se', password: '$2a$10$rTHP5cqRM28MX7kUiuiHPeV6kIK1GTwZG3mbKK7/phDJlB2tQKgua', __v: 0 })

(node:24260) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): ReferenceError: debug is not defined
(node:24260) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

My schema looks like this;

```
const AccessTokenSchema   = new Schema({

  accessToken: { type: String },
  accessTokenExpiresAt: { type: Date, },
  user: { type: Schema.Types.ObjectId, ref: 'User' },
  client: { type: Schema.Types.ObjectId, ref: 'Client' }

});
```

````
const UserSchema = new Schema({

  username: { type: String, unique: true, required: true },
  password: { type: String, required: true },
  admin: { type: Boolean, default: false },
  scope: { type: String }

});
```

```
const ClientSchema = new Schema({

  name: { type: String },
  clientId: { type: String },
  clientSecret: { type: String },
  redirectUris: { type: [String] },
  grants: { type: [String], default: ['authorization_code', 'password', 'refresh_token', 'client_credentials'] },
  scope: { type: String },
  user: { type: Schema.Types.ObjectId, ref: 'User' },

});
```

**Comment by vkarpov15** [comment]
The promise rejection error says "debug is not defined". Are you sure you've done `const debug = require('debug')` in this script?

**Comment by HenrikGr** [comment]
Hi, this can be closed since it works now.

What was missing in the description was that I did not post my code that logged this line;

Mongoose: users.insert({ admin: false, _id: ObjectId("5a82825b82b5ad5ec43b3566"), username: 'henrik@test.se', password: '$2a$10$rTHP5cqRM28MX7kUiuiHPeV6kIK1GTwZG3mbKK7/phDJlB2tQKgua', __v: 0 })

I totally missed that and since I lack knowledge about mongoose. I thougt that was related to the populate function and not to my endpoint code (the part I did not include).

What was even more strange is that I did not use debug in that code. I just recently removed it and my IDE must have cached it or something. :-)

Thanks anyway for this fun library, I enjoying learning more and more.
